libs/pika/execution/include/pika/execution/algorithms/detail/helpers.hpp

```diff
@@ -7,12 +7,33 @@
 #pragma once
 
 #include <pika/config.hpp>
+#include <pika/execution_base/receiver.hpp>
 #include <pika/type_support/pack.hpp>
 
 #include <type_traits>
 #include <variant>
 
+namespace pika::detail {
```

Comment:
Let's maybe keep this in the `pika::execution::experimental::detail` namespace (as for `single_result`)? I think the struct name needs to be a bit more explicit if it's going to be in the top-level `pika::detail` at some point. https://github.com/pika-org/pika/issues/448 is relevant, but no need to mix that into this PR.